docs/source/annotations.rst

```diff
@@ -0,0 +1,68 @@
+****************
+Type Annotations
+****************
+
+Functions
+=========
+
+Parameters
+----------
+
+Asynchronous Functions
+----------------------
+
+Generators
+----------
+
+Lambdas
+-------
+
+
+Classes
+=======
+
+Overloads and Overrides
```

Comment:
Overloads feels more functions-y, is overrides meant to talk about Liskov Substitution Principle? Potentially worth separating